specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2022-12-27-preview/HybridCompute.json

```diff
@@ -0,0 +1,1865 @@
+            "full",
+            "monitor"
+          ],
+          "x-ms-enum": {
+            "name": "agent configuration mode",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "Configurable properties that the user can set locally via the azcmagent config command, or remotely via ARM."
+    },
+    "ConfigurationExtension": {
+      "type": "object",
+      "properties": {
+        "publisher": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Publisher of the extension."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of the extension."
+        }
+      },
+      "description": "Describes properties that can identify extensions."
+    },
+    "ServiceStatuses": {
+      "type": "object",
+      "properties": {
+        "extensionService": {
+          "$ref": "#/definitions/ServiceStatus",
+          "description": "The state of the extension service on the Arc-enabled machine."
+        },
+        "guestConfigurationService": {
+          "$ref": "#/definitions/ServiceStatus",
+          "description": "The state of the guest configuration service on the Arc-enabled machine."
+        }
+      },
+      "description": "Reports the state and behavior of dependent services."
+    },
+    "ServiceStatus": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "type": "string",
+          "description": "The current status of the service."
+        },
+        "startupType": {
+          "type": "string",
+          "description": "The behavior of the service when the Arc-enabled machine starts up."
+        }
+      },
+      "description": "Describes the status and behavior of a service."
+    },
+    "CloudMetadata": {
+      "type": "object",
+      "properties": {
+        "provider": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Specifies the cloud provider (Azure/AWS/GCP...)."
+        }
+      },
+      "description": "The metadata of the cloud environment (Azure/GCP/AWS/OCI...)."
+    },
+    "AgentUpgrade": {
+      "type": "object",
+      "properties": {
+        "desiredVersion": {
+          "type": "string",
+          "description": "Specifies the version info w.r.t AgentUpgrade for the machine."
+        },
+        "correlationId": {
+          "type": "string",
+          "description": "The correlation ID passed in from RSM per upgrade."
+        },
+        "enableAutomaticUpgrade": {
+          "type": "boolean",
+          "description": "Specifies if RSM should try to upgrade this machine"
+        },
+        "lastAttemptTimestamp": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Timestamp of last upgrade attempt"
+        },
+        "lastAttemptStatus": {
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Empty",
```

Comment:
Do we need this "Empty" enum?

Reply:
I believe it should be either success or failure case itself. Maybe just remove that enum ?

Reply:
but then within the design docs, it does have empty as a valid option. <html>
Properties.agentUpgrade.lastAttemptStatus | string | Empty, Success, or Failed

Reply:
well, empty as in null or the empty string or the field isn't provided.
I don't think we need an actual empty enum